### DISCUSSION TITLE
Adding a delta column or subtraction

### DISCUSSION THREAD

**Discussion by jftuga** [grafana/grafana-csv-datasource#55] [Q&A]
If I have this CSV data:

```
Date, Processed
2021-04-15, 300
2021-04-14, 275
```

It would render to this:

| Date | Processed |
|-------|------------|
| 2021-04-15 | 300 |
| 2021-04-14 | 275 |

Would it be possible to use a `transform` or something else to create this:

| Date | Processed | Delta |
|-------|------------|------- |
| 2021-04-15 | 300 | 25
| 2021-04-14 | 275 | |

If that is not possible, would this be:

| Date | Processed |
|-------|------------|
| 2021-04-15 | 25 |
| 2021-04-14 | |

Also, thank you for your excellent work on this plug-in. I have found it very useful.

**Comment by marcusolsson** [comment]
Thank you! While I definitely see the use for this, this is unfortunately outside the scope of the plugin for now. I think this should idea be implemented in Grafana rather than in the data source, as many data sources would benefit from this. 

Until 1.0.0, the focus is on getting the UX of the query editor down, as well as making it stable. That being said, I have been thinking about using https://github.com/mithrandie/csvq to offer a SQL-like query language for CSV data, that could enable operations like this.

**Reply by jftuga** [reply]
I'll do some more Grafana research and see is there is another solution.  You make a good point about this be useful for many data sources which makes me think that someone else might have already worked on this problem.  I will also investigate `csvq` -- I did not know about this.

**Reply by jftuga** [reply]
@marcusolsson 

`csvq` looks to be a perfect solution for my problem as it has the SQL `LAG` function. Thank you for this suggestion.

```
(using a file named test.csv)
csvq 'select Date, Processed, LAG(Processed) over (order by Date) as Previous, (Processed-Previous) as Delta from test order by Date desc'
```

|    Date    | Processed | Previous | Delta |
|------------|-----------|----------|-------|
| 2021-04-15 |  300      |  275     |    25 |
| 2021-04-14 |  275      |   NULL   | NULL  |

This would be wonderful to integrate into your plug-in.

**Reply by marcusolsson** [reply]
Thanks for confirming! If this is important to you, could you maybe create a feature request for it, so that I can add it to the [plugin project board](https://github.com/users/marcusolsson/projects/1)?
